app/client/src/utils/WorkerUtil.ts

```diff
@@ -142,23 +143,37 @@ export class GracefulWorkerService {
      */
     const requestId = `${method}__${_.uniqueId()}`;
     this._channels[requestId] = channel();
+    const mainThreadStartTime = performance.now();
     this._evaluationWorker.postMessage({
       method,
       requestData,
       requestId,
     });
     // The `this._broker` method is listening to events and will pass response to us over this channel.
     const response = yield take(this._channels[requestId]);
-    yield this._channels[requestId].close(); // cleanup
+    const { timeTaken, ...responseData } = response;
+    // Log perf of main thread and worker
+    const mainThreadEndTime = performance.now();
+    const timeTakenOnMainThread = (
+      mainThreadEndTime - mainThreadStartTime
+    ).toFixed(2);
+    const transferTime = (
+      parseFloat(timeTakenOnMainThread) - parseFloat(timeTaken)
+    ).toFixed(2);
+    log.debug(`Worker ${method} took ${timeTaken}ms`);
+    log.debug(`Main ${method} took ${timeTakenOnMainThread}ms`);
+    log.debug(`Transfer ${method} took ${transferTime}ms`);
```

Comment:
```suggestion
    log.debug(`[GracefulWorkerService] Worker ${method} took ${timeTaken}ms`);
    log.debug(`[GracefulWorkerService] Main ${method} took ${timeTakenOnMainThread}ms`);
    log.debug(`[GracefulWorkerService] Transfer ${method} took ${transferTime}ms`);
```

Identifier to easily filter.

Do check `getLogger` on the npm library if it does a cleaner namespacing.